core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -671,22 +690,36 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
     message: () => String,
     cause: Cause[Any],
     overrideLogLevel: Option[LogLevel],
-    trace: Trace
+    trace: Trace,
+    annotations: Map[String, String] = Map.empty[String, String]
   )(implicit unsafe: Unsafe): Unit = {
     val logLevel =
       if (overrideLogLevel.isDefined) overrideLogLevel.get
       else getFiberRef(FiberRef.currentLogLevel)
 
-    val spans       = getFiberRef(FiberRef.currentLogSpan)
-    val annotations = getFiberRef(FiberRef.currentLogAnnotations)
-    val loggers     = getLoggers()
-    val contextMap  = getFiberRefs()
+    val spans        = getFiberRef(FiberRef.currentLogSpan)
+    var annotations0 = getFiberRef(FiberRef.currentLogAnnotations)
+    if (annotations ne Map.empty[String, String]) annotations0 = annotations0 ++ annotations
+
+    val loggers    = getLoggers()
+    val contextMap = getFiberRefs()
 
     loggers.foreach { logger =>
-      logger(trace, fiberId, logLevel, message, cause, contextMap, spans, annotations)
+      logger(trace, fiberId, logLevel, message, cause, contextMap, spans, annotations0)
     }
   }
 
+  private[zio] def logOp(
+    message: String,
+    flags: RuntimeFlags,
+    trace: Trace = Trace.empty,
+    cause: Cause[Any] = Cause.empty
+  )(
+    opLogTrace: Trace
+  ): Unit =
+    if (RuntimeFlags.isEnabled(flags)(RuntimeFlag.OpLog))
```

Comment:
I understand the desire to put the conditional here, because it reduces code duplication. But it's faster to put the conditional at the call site, and make this `logOp` unconditional: if called, it will always log.